lib/datadog/statsd/timer.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Datadog
+  class Statsd
+    class Timer
+      def initialize(interval, &callback)
+        @mx = Mutex.new
+        @cv = ConditionVariable.new
+        @interval = interval
+        @callback = callback
+        @stop = true
+      end
+
+      def start
+        return unless stop?
+
+        @stop = false
+        @thread = Thread.new do
+          last_execution_time = current_time
+          @mx.synchronize do
+            until @stop
+              @cv.wait(@mx, @interval - (current_time - last_execution_time))
```

Comment:
In the case that `current_time - last_execution_time` is > `@interval`, we would be left with a negative time to wait here. The chance for this happening are quite low, but I think that adding a test for this could be worth it.

Reply:
Thank you for pointing out it! I've never thought of such a case. I've changed the code a little to support it in the commit https://github.com/DataDog/dogstatsd-ruby/pull/231/commits/4723e3f11c4b333a4cf2ac7bd325ca58f27e98de.